src/Mapbender/CoreBundle/Resources/public/mapbender.element.scaleselector.js

```diff
@@ -1,65 +1,152 @@
 (function($) {
 
     $.widget("mapbender.mbScaleSelector", {
-        options:      {
+
+        options: {
+            /**
+             * Target widget id string
+             */
             target: null
         },
-        elementUrl:   null,
-        _create:      function() {
+
+        elementUrl: null,
+
+        /**
+         * Map target instance of OpenLayers.Map
+         *
+         * @var {OpenLayers.Map}
+         */
+        olMap: null,
+
+        /**
+         * Mapbender map widget
+         */
+        map: null,
+
+        /**
+         * Ready event listeners
+         *
+         * @var {Array<Function>}
+         */
+        readyCallbacks: [],
+
+        /**
+         * Constructor
+         *
+         * @private
+         */
+        _create: function() {
             var widget = this;
-            if(!Mapbender.checkTarget("mbScaleSelector", widget.options.target)) {
+            var options = widget.options;
+            var target = options.target;
+
+            if(!Mapbender.checkTarget("mbScaleSelector", target)) {
                 return;
             }
+
             var element = $(widget.element);
+
             widget.elementUrl = Mapbender.configuration.elementPath + element.attr('id') + '/';
-            Mapbender.elementRegistry.onElementReady(widget.options.target, $.proxy(widget._setup, widget));
+            Mapbender.elementRegistry.onElementReady(target, $.proxy(widget._setup, widget));
         },
-        _setup:       function() {
+
+        _setup: function() {
             var widget = this;
             var options = widget.options;
-            var mbMap = $('#' + options.target).data('mapbenderMbMap');
-            var scale = mbMap.model.getScale();
-            var scales = mbMap.scales();
+            var mbMap = widget.map = $('#' + options.target).data('mapbenderMbMap');
+            var olMap = widget.olMap = mbMap.map.olMap;
+            var scale = Math.round(olMap.getScale());
             var select = $("select", $(widget.element));
-            $.each(scales, function(idx, value) {
-                value = Math.round(value);
-                select.append('<option value="' + value + '">' + value + '</option>');
+
+            _.each(mbMap.scales(), function(value) {
+                var option = $("<option/>");
+                var formattedValue = Math.round(value);
+
+                option.attr('value', formattedValue).html(formattedValue);
+                select.append(option);
             });
 
             select.change($.proxy(widget._zoomToScale, widget));
-            $('option[value="' + scale + '"]', select).attr('selected', true);
+            select.val(scale);
```

Comment:
You set the value of the select with "scale" which come from olMap (line 58) But the options from select are from the mbMap scales. (line 61 ff.). Why not consistent?

Reply:
@AxxL, as you said options from select are "options from select". `scale` - is current value from OL Map which is set by own element configuration. `options` has any values. Select has this `options` and need current `scale` value to show. `olMap` - is an object to be manipulated by select. mbMap.scales() - are option values from map element configuration. What is wrong?